misc/ls-extensions/modules/partial-parser/src/main/java/io.ballerina.parsers/PartialParserService.java

```diff
@@ -92,4 +106,17 @@
     public String getName() {
         return Constants.CAPABILITY_NAME;
     }
+
+    private String getFormattedSource(String statement) {
+
+        String formattedSourceCode = statement;
+
+        try {
+            formattedSourceCode = Formatter.format(statement);
+        } catch (FormatterException e) {
+            this.clientLogger.notifyUser("Formatting", e);
```

Comment:
`notifyUser()` internally uses `ExtendedLanguageClient#showMessage()` which will show an error popup in VSCode. Is that the correct UX here? What if we just log the error message since you are anyway continuing even after the failure.